### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by ViktoriiaLozova**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <artifactId>qa2018</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>7</source>
+                    <target>7</target>
```
Comment (AntonApanasovych) [inline pom.xml]: why 7th version, use 8

File: src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Root logger option
+log4j.rootLogger=INFO, file, stdout
+
+# Direct log messages to a log file
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.File=C:\\Users\\Simba\\IdeaProjects\\qavlozova2018\\src\\main\\resources\\logging.log
```
Comment (AntonApanasovych) [inline src/main/resources/log4j.properties]: Do not use absolute path. Not everybody have same folder structure as you are. I believe if you simply define a name then logger will create log file in project root.

File: src/main/java/logging/Statistics.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (personCount >= 11) {
+            logger.info("That's enough people for football team");
+        } else {
+            logger.info("That's not enough people for football team");
+        }
+    }
+
+    private static void setupLog4J(){
+        try {
+            System.setProperty("log4j.configuration", new File("src/main/resources", "log4j.properties").toURL().toString());
```
Comment (AntonApanasovych) [inline src/main/java/logging/Statistics.java]: Use `PropertyConfigurator`.
https://howtodoinjava.com/log4j/how-to-configure-log4j-using-properties-file/

File: src/main/java/logging/Person.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    private String name;
+    private String surname;
+    private String email;
+    private ChildCount childCount;
+    private Pet pet;
+
+    private static Logger logger;
+
+    enum ChildCount {
+        One,
```
Comment (AntonApanasovych) [inline src/main/java/logging/Person.java]: enum values are constants and they must follow convention UPPER_CASE_UNDERSCORE

File: src/main/java/logging/Person.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    public Pet getPet() {
+        return pet;
+    }
+
+    public void setPet(Pet pet) {
+        this.pet = pet;
+    }
+
+    public void print() {
+        System.out.println("Name: " + name + " Surname: " + surname + " Email: " + email);
```
Comment (AntonApanasovych) [inline src/main/java/logging/Person.java]: Use logger ;)

File: pom.xml (hunk 1 above)
Comment (ViktoriiaLozova) [reply pom.xml]: Done

File: src/main/resources/log4j.properties (hunk 2 above)
Comment (ViktoriiaLozova) [reply src/main/resources/log4j.properties]: Done

File: src/main/java/logging/Statistics.java (hunk 3 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Statistics.java]: Done

File: src/main/java/logging/Person.java (hunk 4 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Person.java]: Done

File: src/main/java/logging/Person.java (hunk 5 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Person.java]: Done =)

File: src/main/java/logging/Person.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        } else {
+            logger.warn("Not valid pet");
+        }
+
+        Statistics.incrementPersonCount();
+    }
+
+    private static void setupLog4J(){
+        PropertyConfigurator.configure("src/main/resources/log4j.properties");
+        logger = Logger.getLogger(Person.class.getName());
```
Comment (AntonApanasovych) [inline src/main/java/logging/Person.java]: I think you don't need `.getName()`, it is enough to use this way - `Logger.getLogger(Person.class);`

File: src/main/java/logging/Statistics.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        if (personCount >= 11) {
+            logger.info("That's enough people for football team");
+        } else {
+            logger.info("That's not enough people for football team");
+        }
+    }
+
+    private static void setupLog4J(){
+        PropertyConfigurator.configure("src/main/resources/log4j.properties");
+        logger = Logger.getLogger(Statistics.class.getName());
```
Comment (AntonApanasovych) [inline src/main/java/logging/Statistics.java]: remove `getName()`

File: src/main/java/logging/Person.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        DOG,
+        FISH
+    }
+
+    public Person(String name, String surname, String email, int childcount, String pet) {
+        setupLog4J();
+        this.name = name;
+        this.surname = surname;
+        this.email = email;
+        if (childcount == 1){
```
Comment (AntonApanasovych) [inline src/main/java/logging/Person.java]: Hard to read, constructor is too big.
Create private methods `int countChildren(int childcount)` and `String whoIsPet(String pet)`.
Also pet =null will throw NPE

File: src/main/java/logging/Person.java (hunk 6 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Person.java]: Done

File: src/main/java/logging/Statistics.java (hunk 7 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Statistics.java]: Done

File: src/main/java/logging/Person.java (hunk 8 above)
Comment (ViktoriiaLozova) [reply src/main/java/logging/Person.java]: Done

**Pull request merged**